src/shims/unix/env.rs

```diff
@@ -274,12 +275,23 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let this = self.eval_context_mut();
         this.assert_target_os_is_unix("getpid");
 
-        this.check_no_isolation("`getpid`")?;
-
         // The reason we need to do this wacky of a conversion is because
         // `libc::getpid` returns an i32, however, `std::process::id()` return an u32.
         // So we un-do the conversion that stdlib does and turn it back into an i32.
         #[allow(clippy::cast_possible_wrap)]
-        Ok(std::process::id() as i32)
+        Ok(get_pid(this, "`getpid`") as i32)
+    }
+
+    fn gettid(&mut self) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_ref();
+        this.assert_target_os_is_unix("gettid");
+
+        let index = this.machine.threads.active_thread().to_u32();
+
+        #[allow(clippy::arithmetic_side_effects)]
```

Comment:
This is suspicious, why did you allow the lint? It exists for a reason!

Reply:
From what I understand [arithmetic_side_effects](https://rust-lang.github.io/rust-clippy/master/index.html#/arithmetic_side_effects) this lint tries to warn about potential panics or overflows. Since From what I understand the thread IDs and the process IDs come from the same namespace, if it is safe to cast a pid from u32 to i32, `pid + thread_id` would not overflow a u32.

Reply:
We always use `strict_add` (or the equivalent) in Miri unless there's a good reason to use anything else.

Reply:
done.